File: codeforces/1343/C.py
for _ in range(int(input())):
    n = int(input())
    lis = list(map(int,input().split()))
    sum = 0
    i=0
    while(i>=0 and i<n):
        j = i
        cur = lis[i]
        while(j<n and abs(lis[i]+lis[j])==abs(lis[i])+abs(lis[j])):
            cur = max(cur,lis[j])
            j+=1
        sum += cur
        i = j
    print(sum)

File: codeforces/1343/C.cpp
#include<bits/stdc++.h>
using namespace std;
auto sgn = [&](int x) {
		if (x > 0) return 1;
		else return -1;
	};
int main(){
    int T;
    cin>>T;
    while(T--){
        int n;
        cin>>n;
        vector<int> a(n);
        for (int i=0;i<n;++i) cin >> a[i];
        long long sum = 0;
        for(int i=0;i<n;++i){
            int cur = a[i];
            int j = i;
            while (j < n && sgn(a[i]) == sgn(a[j])) {
				cur = max(cur, a[j]);
				++j;
			}
			sum += cur;
			i = j - 1;
        }
        cout << sum << "\n";
    }
    return 0;
}